core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -259,7 +259,8 @@ public void push(Repo<T> repo) throws StackOverflowException {
         throw new StackOverflowException("Repo stack size too large");
       }
 
-      FateMutator<T> fateMutator = newMutator(fateId);
+      FateMutator<T> fateMutator =
+          newMutator(fateId).requireStatus(TStatus.IN_PROGRESS, TStatus.NEW);
```

Comment:
Should not push to something with a NEW status, when something is NEW its never been seeded with an operation.  Was something failing w/o NEW?

```suggestion
          newMutator(fateId).requireStatus(TStatus.IN_PROGRESS);
```

Reply:
> Was something failing w/o NEW?

Yes. All 3 test cases in FateStoreIT fail when run against AccumuloStore when `NEW` is not present here.

Reply:
ok, for that case we could modify the test to set the expected status before calling push

Reply:
Unlike setting the status to make sure the checks pass for the other operations (pop and delete), removing `NEW` from this part of the code cause an error in the table create operation:
```
org.apache.accumulo.core.client.AccumuloException: Internal error processing executeFateOperation

	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:437)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:387)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doTableFateOperation(TableOperationsImpl.java:1737)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.create(TableOperationsImpl.java:275)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.create(TableOperationsImpl.java:245)
	at org.apache.accumulo.test.fate.accumulo.AccumuloStoreFateIT.executeTest(AccumuloStoreFateIT.java:45)
	at org.apache.accumulo.test.fate.accumulo.FateStoreIT.testDeferredOverflow(FateStoreIT.java:149)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.thrift.TApplicationException: Internal error processing executeFateOperation
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:81)
	at org.apache.accumulo.core.manager.thrift.FateService$Client.recv_executeFateOperation(FateService.java:135)
	at org.apache.accumulo.core.manager.thrift.FateService$Client.executeFateOperation(FateService.java:116)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.executeFateOperation(TableOperationsImpl.java:312)
	at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:400)
	... 12 more
```
I am not too sure why this is happening yet, still looking into it.

Reply:
> Caused by: org.apache.thrift.TApplicationException: Internal error processing executeFateOperation

When the above happens that indicates a server side exception happened.  Do you see anything in the manager log?

Reply:
```
java.lang.IllegalStateException: Failed to write mutation REJECTED org.apache.accumulo.core.data.ConditionalMutation@36923a7c
	at org.apache.accumulo.core.fate.accumulo.FateMutatorImpl.mutate(FateMutatorImpl.java:170) ~[classes/:?]
	at org.apache.accumulo.core.fate.accumulo.AccumuloStore$FateTxStoreImpl.push(AccumuloStore.java:264) ~[classes/:?]
	at org.apache.accumulo.core.fate.WrappedFateTxStore.push(WrappedFateTxStore.java:45) ~[classes/:?]
	at org.apache.accumulo.core.logging.FateLogger$LoggingFateTxStore.push(FateLogger.java:57) ~[classes/:?]
	at org.apache.accumulo.core.fate.Fate.seedTransaction(Fate.java:347) ~[classes/:?]
	at org.apache.accumulo.manager.FateServiceHandler.executeFateOperation(FateServiceHandler.java:254) ~[classes/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
	at org.apache.accumulo.core.trace.TraceUtil.lambda$wrapService$0(TraceUtil.java:202) ~[classes/:?]
	at jdk.proxy2/jdk.proxy2.$Proxy27.executeFateOperation(Unknown Source) ~[?:?]
	at org.apache.accumulo.core.manager.thrift.FateService$Processor$executeFateOperation.getResult(FateService.java:561)
```

Reply:
Ok , so looking at Fate.seedTransaction() it pushes while the status is NEW and then later sets the status to SUBMITTED in the same method. So it makes sense to allow NEW for push.